.preview/src/modules/knobs/store.jsx

```diff
@@ -0,0 +1,57 @@
+import { useReducer } from 'react';
+import { without } from 'lodash';
+import updeep from 'updeep';
+
+const DEFAULT_STATE = {
+  entities: {
+    knobs: {},
+    values: {},
+  },
+  knobs: [],
+};
+
+export const TYPE_REGISTER = 'register';
+export const TYPE_UNREGISTER = 'unregister';
+export const TYPE_UPDATE_VALUE = 'updateValue';
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case TYPE_REGISTER: {
+      const { id, defaultValue, ...options } = action.payload;
+
+      const knob = { id, defaultValue, ...options };
+
+      return updeep({
```

Comment:
not immer?

Reply:
I like updeep better for array stuff (maybe because I am more used to it)